### ISSUE TITLE
error when using process substitution as read input

### ISSUE THREAD

**Issue by alshai** [BenLangmead/bowtie2#234]
I'm getting an error when using bash process substitution `<()` syntax to use as input for the read files
Bowtie2 version: 2.3.4.3
```
$ ./bowtie2-2.3.4.3-linux-x86_64/bowtie2 -x data/chr21.fa -1 <(seqtk sample -s 1 data/HG00096.11.fq 0.5) -2 <(seqtk sample -s 1 data/HG00096.12.fq 0.5)
Error: reads file does not look like a FASTQ file terminate called after throwing an instance of 'int'
(ERR): bowtie2-align died with signal 6 (ABRT) (core dumped)
````

It works fine when I use an older version of bowtie2 (2.3.4)

EDIT: I found that the earliest version with which this fails is 2.3.4.2

**Comment by ch4rr0** [comment]
I pushed a fix for this issue that will be included in an upcoming release. Thank you for pointing this out.

**Comment by alshai** [comment]
Thanks!

**Comment by ch4rr0** [comment]
This change is available in v2.4.0 and later.
